hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -308,6 +316,10 @@ public TransactionContext preAppendTransaction(TransactionContext trx)
       CompletableFuture<Message> ratisFuture =
           new CompletableFuture<>();
       applyTransactionMap.put(trxLogIndex, trx.getLogEntry().getTerm());
+
+      //if there are too many pending requests, wait for doubleBuffer flushing
+      availPendingRequestNum.acquire();
+
```

Comment:
It's good practice to keep the Semaphores private and have acquire and release methods for them. It helps in keeping track of where these objects are used. 
Instead of initializing the Semaphore in OzoneManagerStateMachine and passing it over to OzoneManagerDoubleBuffer, can we keep the Semaphores private in OzoneManagerDoubleBuffer and access them via public methods.